include/dma.h

```diff
@@ -270,6 +288,36 @@ static inline int z_impl_dma_stop(struct device *dev, u32_t channel)
 	return api->stop(dev, channel);
 }
 
+/**
+ * @brief get current runtime status of DMA transfer
+ *
+ * Implementations must check the validity of the channel ID passed in and
+ * return -EINVAL if it is invalid.
+ *
+ * @param dev     Pointer to the device structure for the driver instance.
+ * @param channel Numeric identification of the channel where the transfer was
+ *                being processed
+ * @param stat   a non-NULL dma_status object for storing DMA status
+ *
+ * @retval non-negative if successful.
+ * @retval Negative errno code if failure.
+ */
+__syscall int dma_get_status(struct device *dev, u32_t channel,
```

Comment:
I think if you tag this as a syscall, you also need to implement the corresponding [handler](https://docs.zephyrproject.org/latest/reference/usermode/syscalls.html#handler-declaration) in `dma_handlers.c` for it to work with `CONFIG_USERSPACE` set.  I think all you need is the device check like the existing start/stop ones and a `Z_SYSCALL_MEMORY_WRITE` for the status object.  I'm not hugely familiar with the details of the userspace API, however.